cmd/symbols/internal/api/handler_test.go

```diff
@@ -66,7 +66,8 @@ func TestHandler(t *testing.T) {
 		}
 		return newMockParser(pathToEntries), nil
 	}
-	parserPool, err := parser.NewParserPool(parserFactory, 15, parser.DefaultParserTypes)
+	observationCtx := &observation.TestContext
+	parserPool, err := parser.NewParserPool(observationCtx, "test:", parserFactory, 15, parser.DefaultParserTypes)
```

Comment:
rather use TestContextTB, it does validation on the metrics as well and has better reporting. Also a colon in a prometheus metric name probably is not allowed (but ends up working since you used TestContext)
```suggestion
	parserPool, err := parser.NewParserPool(observation.TestContextTB(t), "test", parserFactory, 15, parser.DefaultParserTypes)
```

Reply:
Thanks!